File: 0041-first-missing-positive/0041-first-missing-positive.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int firstMissingPositive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            numSet.add(num);
        }
        
        int target = 1;
        while (numSet.contains(target)) {
            target++;
        }
        
        return target;
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        sor_list = sorted(list(set(nums)))
        
        target = 1
        
        for val in sor_list:
            if val <= 0:
                continue
            
            if val == target:
                target += 1
            elif val > target:
                return target
                
        return target